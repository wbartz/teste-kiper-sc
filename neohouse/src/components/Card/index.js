import PropTypes from 'prop-types';
import React from 'react';
import ReactTooltip from "react-tooltip";
import './index.scss';

const Card = ({ onEdit, id, title, icon, footer }) => {
  return (
    <div className="card">
      <ReactTooltip key={id} />
      <div className="card-header-image">
        <div className="icon">
          <i className="material-icons">{icon}</i>
        </div>
      </div>
      <div className="card-header-edit">
        <i
          onClick={onEdit}
          className="material-icons"
          data-tip="Ver apartamentos"
        >
          visibility
        </i>
      </div>
      <div className="card-body">
        <div className="title">{title}</div>
      </div>
      <div className="footer">{footer}</div>
    </div>
  );
};

Card.propTypes = {
  onEdit: PropTypes.func.isRequired,
  icon: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  footer: PropTypes.node,
};

Card.defaultProps = {
  footer: null,
};

export default Card;