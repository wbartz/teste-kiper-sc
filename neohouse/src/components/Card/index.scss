@import '../../styles/variables';

.card {
  border-radius: 4px;
  width: 306px;
  height: 238px;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 12px 0px;
  margin-bottom: 60px;

  &:hover .card-header-edit {
    visibility: visible;
    transition: visibility 0s linear 300ms, opacity 300ms;
  }

  &:hover .card-header-image {
    margin-top: -75px;
  }

  .card-header-edit {
    cursor: pointer;
    padding: 2.5rem 0rem 0rem !important;
    visibility: hidden;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 30px;
    margin-bottom: 20px;

    i {
      color: $color-secondary !important;
      font-size: 30px !important;
    }
  }

  .card-header-image {
    height: 137px;
    margin-top: -50px;
    border-radius: 5px;
    position: absolute;
    width: 92%;
    transition: margin 0.3s linear;
    box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16),
      0 2px 10px 0 rgba(0, 0, 0, 0.12);
    background-color: #fff;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .card-body {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  i.material-icons {
    font-size: 60px;
    color: $color-lighter-gray;
  }

  .title {
    font-weight: bold;
    font-size: 20px;
    color: $color-primary;
  }
}