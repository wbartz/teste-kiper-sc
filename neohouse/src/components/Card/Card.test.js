import { shallow } from 'enzyme';
import React from 'react';
import Card from '.';

describe('Card', () => {
  const props = {
    onEdit: jest.fn(),
    id: 'teste',
    title: 'teste',
    icon: 'edit',
  };

  it('render without crashing', () => {
    expect(shallow(<Card {...props} />)).toMatchSnapshot();
  });
});
