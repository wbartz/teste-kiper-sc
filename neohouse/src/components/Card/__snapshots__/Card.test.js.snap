// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render without crashing 1`] = `ShallowWrapper {}`;
